### PR TITLE
Taxonomy validator implementation [SUBS-486]

### PR DESCRIPTION
**Pull request by fpenim**

### CODE REVIEW

**Comment by fpenim** [comment]
The branch name is wrong, this refers to ticket **SUBS-486**
https://www.ebi.ac.uk/panda/jira/browse/SUBS-486

File: src/main/java/uk/ac/ebi/subs/validator/taxon/core/TaxonomyService.java [hunk 1]
```diff
@@ -10,7 +10,7 @@
     private final String ENA_TAXONOMY_SERVICE = "http://www.ebi.ac.uk/ena/data/taxonomy/v1/taxon/";
```
Comment (ke4) [inline src/main/java/uk/ac/ebi/subs/validator/taxon/core/TaxonomyService.java:10]: Do we want to hard code the taxonomy service URL to this class?
Is not it better to externalise it to a config file (properties, yml ...)?

File: src/main/java/uk/ac/ebi/subs/validator/taxon/core/TaxonomyValidator.java [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+            } else {
+                taxonomy = taxonomyService.getTaxonByTaxonomicName(sample.getTaxon());
+                sample.setTaxonId(Long.valueOf(taxonomy.getId())); // Are we supposed to do this ???
+            }
+
+            if(taxonomy.isSubmittable()) {
+                logger.debug(SUCCESS_MESSAGE + " " + taxonomy);
+                return generateSingleValidationResult(
+                        sample,
+                        SUCCESS_MESSAGE,
```
Comment (ke4) [inline src/main/java/uk/ac/ebi/subs/validator/taxon/core/TaxonomyValidator.java:53]: SUCCESS_MESSAGE + " " + taxonomy
or using String.format, but then change the above constants, as well:
private final String SUCCESS_MESSAGE = "Valid taxonomy: %s";
String.format(SUCCESS_MESSAGE, taxonomy)

File: src/test/java/uk/ac/ebi/subs/validator/taxon/TaxonomyValidatorTest.java [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+import uk.ac.ebi.subs.validator.taxon.core.TaxonomyValidator;
+
+import java.util.UUID;
+
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+public class TaxonomyValidatorTest {
+
+    private final String SUCCESS_MESSAGE = "Valid taxonomy";
```
Comment (ke4) [inline src/test/java/uk/ac/ebi/subs/validator/taxon/TaxonomyValidatorTest.java:20]: If you changed the messages in the service, then change it here as well, please.

**Review by ke4 — APPROVED** [review]
Looks good.
I added some comments, but they are not very important to fix.

**Pull request merged**